sdk/python/kfp/components/tasks_group.py

```diff
@@ -137,6 +137,35 @@ def __init__(
 
         self.exit_task = exit_task
 
+    def surface_outputs(self, task):
+        # For channels
+        self._outputs = {
+            task.name: pipeline_channel.create_pipeline_channel(
+                name=channel.name,
+                channel_type=channel.channel_type,
+                task_name=task.name,
+            ) for _, channel in (task.outputs or {}).items()
+        }
+
+    @property
+    def channel_inputs(self) -> List[pipeline_channel.PipelineChannel]:
+        """The list of all channel inputs passed to the task."""
+        return []
+
+    @property
+    def outputs(self) -> Mapping[str, pipeline_channel.PipelineChannel]:
+        """The dictionary of outputs of the task.
+
+        Used when a task has more the one output or uses an
+        ``OutputPath`` or ``Output[Artifact]`` type annotation.
+        """
+        return self._outputs
+
+    @property
+    def dependent_tasks(self) -> List[str]:
+        """A list of the dependent task names."""
+        return []
```

Comment:
I don't have a suggestion for this, but I'm curious if there is a way to implement this without having to add these "dummy" methods, which just make a `TasksGroup` look like a `PipelineTask`? Ideally want to avoid having to add special handling for the ExitHandler case.